crates/red_knot_python_semantic/src/types/call/bind.rs

```diff
@@ -135,20 +153,300 @@ impl<'db> Bindings<'db> {
             first.report_diagnostics(context, node);
         }
     }
+
+    /// Evaluates the return type of certain known callables, where we have special-case logic to
+    /// determine the return type in a way that isn't directly expressible in the type system.
+    fn evaluate_special_cases(&mut self, db: &'db dyn Db) {
```

Comment:
Nit: Should we rename this to `evalute_known_cases` to reuse our existing terminology around known classes, functions?

Reply:
Done